src/snowflake/cli/plugins/object/stage/manager.py

```diff
@@ -53,9 +53,12 @@ def _to_uri(self, local_path: str):
             return uri
         return to_string_literal(uri)
 
-    def list_files(self, stage_name: str) -> SnowflakeCursor:
+    def list_files(self, stage_name: str, pattern: str = ".*") -> SnowflakeCursor:
         stage_name = self.get_standard_stage_name(stage_name)
-        return self._execute_query(f"ls {self.quote_stage_name(stage_name)}")
+        query = f"ls {self.quote_stage_name(stage_name)}"
+        if pattern is not None:
+            query += f" pattern = '{pattern}'"
```

Comment:
What if pattern contains `'`?

I checked and the following does not work:
```
snow object stage list TEST_STAGE_1 -p ".*asd'\.txt"
```

IMO we should either escape it (if possible) or inform users that they have to do it by themselves (if possible, in other case that they cannot use `'`).

Reply:
I think it can be more generic problem in building SQLs in our tool and maybe we should think about it outside of the context of this param. 

But also maybe in other cases `'` char shouldn't be allowed to be used in a value being part of SQL query, so maybe it's not a problem there but here this char is a valid part of filenames:

```
snow object stage list test_stage_1
ls @test_stage_1
+-----------------------------------------------------------------------------------------------------------+
| name                            | size | md5                              | last_modified                 |
|---------------------------------+------+----------------------------------+-------------------------------|
| test_stage_1/'test 1'.txt       | 16   | 0d56227f012d6bf1881bffa3e14d55a6 | Thu, 14 Mar 2024 13:48:05 GMT |
```

Reply:
Personally, I think it's so uncommon char in filenames that mentioning the issue to users should be enough.

Reply:
help message was getting very long, so I decided to add validation and readable error message:
```
╭─ Error ───────────────────────────────────────────────────╮
│ All "'" characters in PATTERN must be escaped: "\'"       │
╰───────────────────────────────────────────────────────────╯
```